// genius-lyrics/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.version = exports.Utils = exports.Constants = exports.Album = exports.Artist = exports.Song = exports.ArtistsClient = exports.SongsClient = exports.Client = void 0;
const Client_1 = require("./Client");
Object.defineProperty(exports, "Client", { enumerable: true, get: function () { return Client_1.Client; } });
const Client_2 = require("./Songs/Client");
Object.defineProperty(exports, "SongsClient", { enumerable: true, get: function () { return Client_2.SongsClient; } });
const Client_3 = require("./Artists/Client");
Object.defineProperty(exports, "ArtistsClient", { enumerable: true, get: function () { return Client_3.ArtistsClient; } });
const Song_1 = require("./Songs/Song");
Object.defineProperty(exports, "Song", { enumerable: true, get: function () { return Song_1.Song; } });
const Artist_1 = require("./Artists/Artist");
Object.defineProperty(exports, "Artist", { enumerable: true, get: function () { return Artist_1.Artist; } });
const Album_1 = require("./Albums/Album");
Object.defineProperty(exports, "Album", { enumerable: true, get: function () { return Album_1.Album; } });
const Constants_1 = require("./Constants");
Object.defineProperty(exports, "Constants", { enumerable: true, get: function () { return Constants_1.Constants; } });
const Utils = __importStar(require("./Utils"));
exports.Utils = Utils;
const version = require("../package.json").version;
exports.version = version;
exports.default = {
    Client: Client_1.Client,
    SongsClient: Client_2.SongsClient,
    ArtistsClient: Client_3.ArtistsClient,
    Song: Song_1.Song,
    Artist: Artist_1.Artist,
    Album: Album_1.Album,
    Constants: Constants_1.Constants,
    Utils,
    version,
};
